src/client/src/models/Scene.js

```diff
@@ -0,0 +1,28 @@
+// This object represents the list of scenes which will be displayed on the scenes page and exported via HTTP to the server.
+// Basically, this is our key "JSON object"
+// We use localStorage for persistence
+
+const Scene = {
+  initialize: function() {
+    // before initializing, make sure there isn't already data
+    if (!localStorage.getItem('scene-list')) {
+      // for now hard-coding in some example scenes
+      localStorage.setItem('scene-list', JSON.stringify([{id: 1, captions: [{id: 1}]}, {id: 2, captions: []}]))
+    }
+  },
+
+  getScenes: function() {
+    return JSON.parse(localStorage.getItem('scene-list'))
+  },
+
+  addScene: function() {
+    // get current list
+    const list = JSON.parse(localStorage.getItem('scene-list'))
```

Comment:
Should this default to an empty list in case we fetch no scenes? something like  `|| []`

Reply:
I don't know... sort of a stylistic preference. 

The way the code is written now, that case will never happen because `initialize` is called first thing, so we know we will always have something. 

So the question is should we add checks that are redundant/unnecessary now, with the idea that in the future when code is changed, those checks might then be useful to have? I could certainly see it happening where we refactor the code in a way so it doesn't call initialize or something, in which case this code will break at that very place, and you will be right to have wanted a check there.

However, I could also see the case that changing the way the app is initialized would be a mistake, and so having the app throw an exception and blow up would inform whoever made that change that they broke some depended on behavior. Whereas if we defaulted to an empty array as you suggest, the app might appear to still work (since it won't throw an exception there), even though it didn't initialize properly. A bug might pop up further along, so ultimately they will have to trace back and figure out the cause of the problem. It might be more difficult to realize that the original culprit had to do with initialization in that case. 

Essentially, premature checks like that can have the consequence of delaying when a problem is encountered, making it harder to trace back the root cause. This is the tradeoff that comes with the fact that the user may be less likely to experience a crash.

So do we want to minimize/delay the chances of a crash by adding redundant checks? Or do we want to let crashes happen as soon as possible so as to more quickly discover a problem?

This is an interesting debate to be had... I'm not sure where I stand myself, to be honest. I think it probably is a spectrum and I have to take things case by case. In this case, however, I think I stand by my original decision. I would rather not hide a problem with initialization. I would want to know right away if I messed that up. I guess I might fall closer to the "blow up sooner rather than later" end of the spectrum.

Reply:
I think when a user loads up the app, there should be an empty list of scenes. As it stands the program is not crashing only because there are a couples scenes hard-coded in. 

I may be misunderstanding, what are we trying to initialize?

Reply:
I wanted there to be a few hard-coded scenes for the sake of demonstration (of the current data format, and to see the scenes on the UI).

Of course at some point we will want to get rid of those hard-coded scenes. And when we do, we will change the initialize function to not hard-code the data, but instead put in an empty list.

So for example, we could change initialize to look something like this:

```
initialize: function() {
    // before initializing, make sure there isn't already data
    if (!localStorage.getItem('scene-list')) {
        // since localStorage is empty, initialize to an empty list
        localStorage.setItem('scene-list', JSON.stringify([]))
    }
},
...
``` 

That would be a proper way to initialize. Imagine a situation where someone wanted to get rid of the hard-coded data, and they noticed that there was a check on addScene like you suggested `... || []`. They might think it is fine to just delete the initialize function altogether, now that hard-coded data is no longer needed, and the addScene function will properly default to an empty list.

The problem with the above scenario is that maybe somewhere the scene list is depended on existing _before_ addScene is ever called. For example, maybe in one of the View components, the scene list is needed to render the list of scenes. An empty list would be fine, but since initialize was never called, and neither was addScene, the scene list might be null or undefined. This would cause that view to crash.

Which leads me back to my original point. "Fixing" the null/undefined situation in the addScene function could just lead to false confidence that the situation has been resolved. Rather, it is better (in my opinion) to structure the code in such a way that we expect that initialize has been called, and we expect initialize to properly set the values (whether that be hard-coded values, a template, or even just an empty list).

If addScene blows up because it was expecting a list but it got undefined/null, then we know there is a problem further back, and we can work on figuring out how that happened.

If instead we add the default case in addScene, we risk just pushing the problem somewhere else, and some other file might blow up.

Reply:
For the record, I enjoy conversations like this, so don't hesitate to push back on anything I say. 

That goes for everyone.

Reply:
I'm understanding better now. The other thing i wasn't too sure about was whether we wanted local storage or session storage. I'm trying to think of being an end-user but i'm torn between the pros/cons. We can discuss this with the sponsors and figure it out at a later time. This all looks good to me, let's ship it!

Reply:
Thanks Marko, I'll merge it.

I chose local storage because I thought if they accidentally closed their browser window or something, they wouldn't want to lose all of their work.